File: LeetCode/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mn, mx = prices[0], 0
        
        for i in range(1,len(prices)):
            mn = min(mn, prices[i])
            mx = max(mx, prices[i]-mn)
        
        return mx

File: LeetCode/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mn = prices[0], mx = 0;
        
        for(int i=1; i < prices.size(); i++) {
            mn = min(mn, prices[i]);
            mx = max(mx, prices[i]-mn);
        }
        
        return mx;
    }
};